src/platform_win.cc

```diff
@@ -351,4 +356,22 @@ optional<std::string> RunExecutable(const std::vector<std::string>& command,
   return output;
 }
 
+optional<std::string> GetGlobalConfigDirectory() {
+	wchar_t  *roaming_path = NULL;
+	optional<std::string> cfg_path = {};
+	if (SUCCEEDED(SHGetKnownFolderPath(FOLDERID_RoamingAppData, KF_FLAG_DEFAULT,
+					   NULL, &roaming_path))) {
+	  std::wstringstream roaming_stream;
+          roaming_stream << roaming_path << L"/cquery";
+
+	  // Convert the roaming path string to a normal string so it
+	  // is analogous with the string returned by the posix version.
+	  using convert_type = std::codecvt_utf8<wchar_t>;
+	  std::wstring_convert<convert_type, wchar_t> converter;
+	  cfg_path = converter.to_bytes(roaming_stream.str());
+	}
+	CoTaskMemFree(static_cast<void*>(roaming_path));
```

Comment:
https://msdn.microsoft.com/en-us/library/windows/desktop/ms680722(v=vs.85).aspx it sounds like this should only be used if the memory was allocated using CoTaskMemAlloc or CoTaskMemRealloc.

Btw, is there a sample somewhere to do this? I'd recommend referencing any doc pages here.

Reply:
In the docs for https://msdn.microsoft.com/en-us/library/bb762188(VS.85).aspx, it says call is responsible for freeing `ppszPath` (in this case `roaming_path`) using `CoTaskMemFree`. I have moved the call inside the block where the we use it because `CoTaskMemFree` does nothing when the parameter is NULL.

Reply:
I have referenced the docs in a comment above the call as you recommended